### ISSUE TITLE
amplify push result in error in Windows 10

### ISSUE THREAD

**Issue by kktam** [aws-amplify/amplify-js#2042]
**Describe the bug**
When the publish the changes using the command `amplify push`. In Windows 10, it produces the following error:

```
C:\Users\<user>\code\aws\amplify\expo-amplify>amplify push
| Category | Resource name   | Operation | Provider plugin   |
| -------- | --------------- | --------- | ----------------- |
| Auth     | <id>| No Change | awscloudformation |
? Are you sure you want to continue? true

Cannot find module '/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation'
```

**To Reproduce**
1. Go to root directory of the amplify project
2. Type 'amplify push'

**Expected behavior**
Configuration published to Cloud formation.

**Screenshots**
As described above.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser Chrome
 - Version 
    Amplify 0.1.31
   Node v10.13.0
   npm 6.4.1

**Smartphone (please complete the following information):**
 - Device: Android on Expo 2
 - OS: Android 7.0 or higher
 - Browser [Chome]

**Additional context**

***You can turn on the debug mode to provide more info for us by setting window.LOG_LEVEL = 'DEBUG'; in your app.***

**Comment by kktam** [comment]
I discovered the root cause of this issue.

I tried to work around a Windows 10 issue to build an Amplify project from Ubuntu and clone it into Windows to work in a mixed environment. Apparently the environment path are coded into Amplify configurations and cannot be shared across different OS platform.

The original Windows 10 blocking issue is [https://github.com/aws-amplify/amplify-cli/issues/355](https://github.com/aws-amplify/amplify-cli/issues/355) and is not resolved by newer amplify-cli version.

Reproduce step for this issue.

1. create new react project and add amplify
2. commit into github repo
3. clone project into Windows 10 client
4. run react project

In the future, will there be support to abstracted out the OS dependent path and allow a "generic" react amplify project run in both Windows and Linux clients?

Also, since newest Windows 10 now support Linux mode [https://docs.microsoft.com/en-us/windows/wsl/install-win10](https://docs.microsoft.com/en-us/windows/wsl/install-win10), isn't there any way to hide the linux code inside Linux mode, so the amplify configuration can be run anywhere?
